### PR TITLE
feat(plugin): add preserve sourcemap option

### PR DESCRIPTION
**Pull request by JonasBa**

### CODE REVIEW

File: src/build-plugin.ts [hunk 1]
```diff
@@ -6,14 +6,15 @@ import { maybeRewriteSourcemapWithReactProd, loadSourcemap } from "./index";
...
   debug?: boolean;
+  preserve?: boolean;
 }
 
 const PLUGIN_NAME = "react-sourcemaps";
 // Iterates over the list of generated assets, skips any that are not sourcemaps
 // and attempts to rewrite them with React production sourcemaps.
 function rewireSourceMapsFromGeneratedAssetList(
   generatedAssets: string[],
-  options: ReactSourcemapsPluginOptions = { debug: false}
```
Comment (JonasBa) [inline src/build-plugin.ts:16]: the default should only be initialized in one place (the unplugin factory) so we avoid multiple initializers and possible subtle bugs

**Review by markerikson — APPROVED** [review]

**Comment by markerikson** [comment]
No real context here, so let's go with it!

**Comment by JonasBa** [comment]
Sorry @markerikson - this option is to preserve the old sourcemaps so we don't end up overriding them when we remap.

**Comment by markerikson** [comment]
Yeah, sorry, I got that part - more saying that my brain is having to context switch a lot, and don't have time or mental space to fully review and analyze the code. I trust your judgment here! :)

**Comment by JonasBa** [comment]
Ah yes, I understand that. I'll make sure I write better context and add comments for any new PR's so that it's easier to review next time. Thanks for merging my PR 🙏🏼

**Comment by markerikson** [comment]
Sure, and thank _you_ for putting in all the effort to help flesh out this package!

Yeah, right now I'm trying to juggle:

- Replay.io day job stuff (currently: writing E2E tests)
  - in my free time, hacking on some experimental features for fun because they legitimately interest me
- Thinking about or actually convincing myself to work on the constellation of RTK 2.0-related tasks
- Actually attempting to have a life and some balance :)

I definitely _wanted_ to have some kind of build tool integration for this package, but I didn't have the time or capacity to work on it myself. So, having you build that out is wonderful!

**Pull request merged**